numba/tests/test_parfors.py

```diff
@@ -2540,6 +2540,21 @@ def test_impl(image, a, b):
         image = np.zeros((3, 3), dtype=np.int32)
         self.prange_tester(test_impl, image, 0, 0)
 
+    @skip_parfors_unsupported
+    def test_list_setitem_hoisting(self):
+        # issue5979
+        # Don't hoist list initialization if list item set.
```

Comment:
Is this comment accurate? It seems like in the above the code is restricting hoisting to `np.empty` as the allocator?

Reply:
The code I added to stop list initialization hoisting when inappropriate (line 768 above) had the side-effect of stopping np.empty from being hoisted.  So, I added something specifically back in for just that function to make the existing hoisting tests pass.  So, this comment is accurate and relates to the core change of this PR on line 768.  Everything besides 768-769 is just in support of that change.

Reply:
Yes, it is accurate.  This was the main change of the PR which had the side-effect of stopping normal np.empty hoisting.  I had to add something back in specifically for that then.

Reply:
ah, great, thanks for the explanation.